pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -264,17 +264,9 @@ public Plan makeStreamingInstancePlan(List<IndexSegment> indexSegments, QueryCon
       planNodes.add(makeStreamingSegmentPlanNode(indexSegment, queryContext));
     }
     CombinePlanNode combinePlanNode = new CombinePlanNode(planNodes, queryContext, executorService, streamObserver);
-    if (QueryContextUtils.isSelectionOnlyQuery(queryContext)) {
-      // selection-only is streamed in StreamingSelectionPlanNode --> here only metadata block is returned.
-      return new GlobalPlanImplV0(
-          new InstanceResponsePlanNode(combinePlanNode, indexSegments, Collections.emptyList(), queryContext));
-    } else {
-      // non-selection-only requires a StreamingInstanceResponsePlanNode to stream data block back and metadata block
-      // as final return.
-      return new GlobalPlanImplV0(
-          new StreamingInstanceResponsePlanNode(combinePlanNode, indexSegments, Collections.emptyList(), queryContext,
-              streamObserver));
-    }
+    return new GlobalPlanImplV0(
+        new StreamingInstanceResponsePlanNode(combinePlanNode, indexSegments, Collections.emptyList(), queryContext,
```

Comment:
Not sure I follow why this gets explicitly hard-coded to Streaming for everything ?

Reply:
correct. we are going to `makeStreamingInstancePlan` all the time. b/c for selection only it is going to stream block by block. and for aggregation we will do segment level combine first then still string data block by block. 

next PR will make this configurable. e.g. do we want to do segment level combine, and how much combine we want to do. but the idea is use streaming to configure it otherwise we are limited to only able to return 1 data block